api/tests/routes/public/helpers.py

```diff
@@ -130,6 +132,34 @@ def setup_active_venue_provider(
         return plain_api_key, venue_provider
 
 
+class PublicAPIRestrictedEnvHelper(PublicAPIEndpointBaseHelper):
+    @override_settings(IS_PROD=True)
+    def test_should_not_be_usable_from_production_env(self, client):
+        plain_api_key, _ = self.setup_provider()
+        authenticated_client = client.with_explicit_token(plain_api_key)
+
+        booking = educational_factories.CollectiveBookingFactory()
+
+        response = self.send_request(
+            authenticated_client, url_params={"booking_id": booking.id}, json={"uai": "123UAI", "redactor_id": 1}
+        )
+
+        assert response.status_code == 403
+        assert "unauthorized action" in response.json["msg"]
```

Comment:
Alors en suivant mon commentaire précédent (c'est à dire en remontant le décorateur `@utils.exclude_prod_environment`) tu peux grandement simplifier le test
```suggestion
    def test_should_not_be_usable_from_production_env(self, client):
        plain_api_key, _ = self.setup_provider()
        authenticated_client = client.with_explicit_token(plain_api_key)

        url = self.endpoint_url

        if self.default_path_params:
            url = url.format(**self.default_path_params)
        
        client_method = getattr(authenticated_client, self.endpoint_method)
        response = client_method(url)

        assert response.status_code == 403
        assert "unauthorized action" in response.json["msg"]
```